### PR TITLE
Reader: Remove align attributes from excerpts

### LINKED ISSUE #13384: Reader Search: Post's text centered
On this search: https://wordpress.com/read/search?q=botany

This post's text is centered: https://wordpress.com/read/blogs/43201448/posts/2248

screen shot 2017-04-25 at 11 23 43 am

@blowery Is this the same problem as https://github.com/Automattic/wp-calypso/issues/13356 ?

### PR DESCRIPTION
**Pull request by blowery**
Fixes #13384

### CODE REVIEW

File: client/lib/post-normalizer/test/index.js [hunk 1]
```diff
@@ -968,7 +968,7 @@ describe( 'index', function() {
 <p>&nbsp;</p>
 <p class="wp-caption">caption</p>
 <p><img src="http://example.com/image.jpg"></p>
-<p><a href="http://wikipedia.org">Giraffes</a> are <br>great</p>
+<p align="left" style="text-align:right"><a href="http://wikipedia.org">Giraffes</a> are <br>great</p>
```
Comment (samouri) [inline client/lib/post-normalizer/test/index.js]: 'strips empty elements, leading brs, and alignment settings'

Comment (blowery) [inline client/lib/post-normalizer/test/index.js]: done

**Comment by samouri** [comment]
Do we also want to strip it for full-post view? If not then 🚢.

Functionality looked good for excerpts in the stream.

**Comment by fraying** [comment]
> Do we also want to strip it for full-post view?

Yes please!

**Comment by blowery** [comment]
> > Do we also want to strip it for full-post view?

> Yes please!

I'm not so sure. There are a number of forms, especially poetry, that rely on centered or otherwise aligned text. If the author added in a text-align or p.align to the actual content, maybe we should respect that?

That said, we currently strip all user styles, so they only case slipping through right now is `p[align]`. Removing that would make things a bit more consistent. 

I see excerpts as a special case where dropping formatting is more acceptable, or at least understandable.

**Comment by blowery** [comment]
Just as a check, if you explicitly align text with the current editor, it uses `text-align` on the `p` tag to handle it. See https://b19ytest.wordpress.com/2017/04/26/left-side-right-side-middle-side/

And we strip it. See https://wordpress.com/read/feeds/25320833/posts/1434642305

**Comment by blowery** [comment]
updated to strip `align` from non-`img` tags in content to be consistent with current practice.

@fraying also looking at being more selective in what we strip in the presence of galleries. There may be some things we can do, but it'll be pretty touchy. Probably take a day of work to write and test.

**Comment by fraying** [comment]
I agree that we should strip the alignment out of excerpts, and that there may be cases where it'd be acceptable on the fullpost. Still, as you said, since we strip other aligns, and these were basically a bug, I'd prefer to remove them from full post too.

File: client/lib/post-normalizer/rule-content-remove-styles.js [hunk 2]
```diff
@@ -28,5 +24,12 @@ export default function removeContentStyles( post, dom ) {
 		element.parentNode && element.parentNode.removeChild( element );
 	} );
 
+	// remove align from non images
```
Comment (samouri) [inline client/lib/post-normalizer/rule-content-remove-styles.js]: why only non images?

**Comment by samouri** [comment]
Code LGTM.

Testing now.

**Pull request merged**